lib/system-nixpkgs-map.nix

```diff
@@ -17,8 +17,13 @@ pkgs:
      GL = pkgs.libGL;
      GLU = pkgs.libGLU;
      alut = pkgs.freealut;
-     X11 = pkgs.xorg.libX11;
+     X11 = pkgs.buildEnv {
+       name = "x11-haskell-wrapper";
+       paths = with pkgs.xorg; [ libX11 libXScrnSaver libXinerama ]
+               ++ [ pkgs.xlibs.libXinerama.dev ];
+     };
```

Comment:
An alternative would be to do this as a nixpkgs overlay. We already have some of those already. So we could globally override `libX11` to include these. Not sure if that's a good idea.

Reply:
Using overlays sounds like a cleaner approach, although it's unclear to me how they can be used in conjunction with flakes.

Reply:
@michaelpj I don't think overlays are necessary here because we need a one-off adjustment to just the pkgs set which is used for resolving cabal system package dependencies.

Reply:
Sounds good, 
this will make Xinerama a hard dependency for X11.

Context:
https://github.com/xmonad/X11/blob/4aa04e20334810cb0c8fcfc19c0789b99920ee21/configure.ac
**Xinerama** seems to be the only _soft_ dependency of xmonad/X11 which is not included in `libX11` or `xorgproto` (which is a `propagatedBuildInput` of `libX11`)
**XScrnSaver** is a hard dependency since https://github.com/xmonad/X11/pull/61